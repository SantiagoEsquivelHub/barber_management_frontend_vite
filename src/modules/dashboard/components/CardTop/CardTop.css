.cardTop {
    border: 1px solid #fdc43f;
    border-radius: 20px;
    margin: 1% 0%;
    background-color: #f6f6f6;
}

.cardTop .ant-avatar {
    width: 40px !important;
    height: 40px !important;
    line-height: 32px !important;
    border-radius: 32% !important;
    margin-left: 10px !important;
}

.cardTop .ant-list-item-meta {
    align-items: center !important;
}