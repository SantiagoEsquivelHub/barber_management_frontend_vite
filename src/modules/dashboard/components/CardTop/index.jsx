import React from 'react'
import iconBarber from '../../../../assets/images/icono.png'
import './CardTop.css';

 const CardTop = ({ nombre, url, numServicios, id }) => {
    return (
        <div>
            <div id={id} className="cardTop">

                <li className="ant-list-item"><div class="ant-list-item-meta">
                    <div className="ant-list-item-meta-avatar">
                        <span className="ant-avatar ant-avatar-circle ant-avatar-image">
                            <img src={url} />
                        </span>
                    </div>
                    <div className="ant-list-item-meta-content">
                        <h4 className="ant-list-item-meta-title" >
                            <a href={`/staff/${id}`} id={id}>{nombre}</a>
                        </h4>
                    </div>
                    <div className="ant-list-item-meta-content">
                        <div className="ant-list-item-meta-description">{numServicios == 1 ? '1 servicio en mes' : `${numServicios} servicios en mes`}</div>
                    </div>
                </div>
                    <ul className="ant-list-item-action">
                        <li>
                            <img src={iconBarber} className="icono_barber"/>
                        </li>
                    </ul>
                </li>
            </div>
        </div>
    )
}
export default CardTop;
