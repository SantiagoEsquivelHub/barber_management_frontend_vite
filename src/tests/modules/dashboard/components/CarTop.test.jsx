import { render } from "@testing-library/react"
import CardTop from "../../../../modules/dashboard/components/CardTop";
import icon from '../../../../assets/images/UserPhoto.png';


describe('Pruebas en <CardTop />', () => {

    test('Debe hacer match con el Snapshot', () => {

        const nombre = 'Santiago';
        const numServicios = 'Santiago';
        const id = 10;

        const { container } = render(<CardTop nombre={nombre} url={icon} numServicios={numServicios} id={id}/>)
        expect(container).toMatchSnapshot();
    })

})