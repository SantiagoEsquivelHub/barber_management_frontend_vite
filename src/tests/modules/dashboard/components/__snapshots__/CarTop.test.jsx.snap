// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CardTop /> Debe hacer match con el Snapshot 1`] = `
<div>
  <div>
    <div
      class="cardTop"
      id="10"
    >
      <li
        class="ant-list-item"
      >
        <div
          class="ant-list-item-meta"
        >
          <div
            class="ant-list-item-meta-avatar"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
            >
              <img
                src=""
              />
            </span>
          </div>
          <div
            class="ant-list-item-meta-content"
          >
            <h4
              class="ant-list-item-meta-title"
            >
              <a
                href="/staff/10"
                id="10"
              >
                Santiago
              </a>
            </h4>
          </div>
          <div
            class="ant-list-item-meta-content"
          >
            <div
              class="ant-list-item-meta-description"
            >
              Santiago servicios en mes
            </div>
          </div>
        </div>
        <ul
          class="ant-list-item-action"
        >
          <li>
            <img
              class="icono_barber"
              src=""
            />
          </li>
        </ul>
      </li>
    </div>
  </div>
</div>
`;
